### ISSUE TITLE
Open network of .xml file does not open it

### ISSUE THREAD

**Issue by SBL2021** [simbrain/simbrain#103]
I followed step by step the creation and saving of 4 small networks. However when I try and open them with open network cmd I do not see any GUI popping up. 
I am running Simbrain3 on windows 10. 
Greatly appreciate a rapid answer

**Comment by jyoshimi** [comment]
What version of java are you using? Issues are arising with newer versions of java and we are working to produce a release that addresses these. For now if you run using java 8 it should work. Amazon's corretto tends to work well: https://docs.aws.amazon.com/corretto/latest/corretto-8-ug/downloads-list.html

**Comment by SBL2021** [comment]
Thanks Jeff for the quick answer.
Here is the version I got: 
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)

**Comment by jyoshimi** [comment]
Did you try saving and reopening when running Simbrain using that version? Hopefully it fixes it. 

If it still breaks, try running Simbrain from the command line.  Open a powershell window, navigate to where the Simbrain.jar file is, and run using "java -jar Simbrain.jar". Then try to save / open and see if any error messages show in the powershell window.  You can then paste those here.

**Comment by SBL2021** [comment]
Yes I did close and repopen Simbrain but it did not work.
Here is what I get when running the cmd line  "java -jar Simbrain.jar"
log4j:WARN No appenders could be found for logger (org.simbrain.workspace.Workspace).
log4j:WARN Please initialize the log4j system properly.
Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
 at org.simbrain.workspace.WorkspaceComponentDeserializer.deserializeWorkspaceComponent(WorkspaceComponentDeserializer.java:97)
        at org.simbrain.workspace.gui.GuiComponent.showOpenFileDialog(GuiComponent.java:172)
        at org.simbrain.workspace.component_actions.OpenAction.actionPerformed(OpenAction.java:66)
        at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
        at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
        at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
        at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
        at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
        at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
        at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
        at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
        at java.awt.Component.processMouseEvent(Component.java:6539)
        at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
        at java.awt.Component.processEvent(Component.java:6304)
        at java.awt.Container.processEvent(Container.java:2239)
        at java.awt.Component.dispatchEventImpl(Component.java:4889)
        at java.awt.Container.dispatchEventImpl(Container.java:2297)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
        at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
        at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
        at java.awt.Container.dispatchEventImpl(Container.java:2283)
        at java.awt.Window.dispatchEventImpl(Window.java:2746)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
        at java.awt.EventQueue$4.run(EventQueue.java:733)
        at java.awt.EventQueue$4.run(EventQueue.java:731)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.simbrain.workspace.WorkspaceComponentDeserializer.deserializeWorkspaceComponent(WorkspaceComponentDeserializer.java:90)
        ... 41 more
Caused by: com.thoughtworks.xstream.converters.ConversionException: Cannot construct org.simbrain.network.core.Neuron as it does not have a no-args constructor : Cannot construct org.simbrain.network.core.Neuron as it does not have a no-args constructor
---- Debugging information ----
message             : Cannot construct org.simbrain.network.core.Neuron as it does not have a no-args constructor
cause-exception     : com.thoughtworks.xstream.converters.reflection.ObjectAccessException
cause-message       : Cannot construct org.simbrain.network.core.Neuron as it does not have a no-args constructor
class               : org.simbrain.network.core.Neuron
required-type       : org.simbrain.network.core.Neuron
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
path                : /org.simbrain.network.core.Network/neuronList/org.simbrain.network.core.Neuron
class[1]            : java.util.ArrayList
converter-type[1]   : com.thoughtworks.xstream.converters.collections.CollectionConverter
class[2]            : org.simbrain.network.core.Network
version             : null

**Comment by jyoshimi** [comment]
Ok, and just to confirm, when you are in the powershell prompt and enter "java -version", it is 1.8. I ask in case the java you installed is not the same as the java actually running Simbrain.

**Comment by SBL2021** [comment]
Yes correct here is thescreen copy
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.292-b10, mixed mode)

**Comment by jyoshimi** [comment]
Can you post the xml or a zip workspace file? I can have a look. Puzzled by this...

**Comment by jyoshimi** [comment]
Also another thing you can try. Just make a simple network with one neuron or something, and try to save and re-open that. If that fails, send me the zip or xml.

**Comment by SBL2021** [comment]
the files are in .xml and seems cannot copy & paste them here I will zip them

**Comment by SBL2021** [comment]
[AND.zip](https://github.com/simbrain/simbrain/files/8741765/AND.zip)
[XOR.zip](https://github.com/simbrain/simbrain/files/8741771/XOR.zip)

**Comment by SBL2021** [comment]
Tried creating a new network and saved it. Trying to reopen does not work

**Comment by SBL2021** [comment]
What is the solution for the warning in the config?
log4j:WARN No appenders could be found for logger (org.simbrain.workspace.Workspace).
log4j:WARN Please initialize the log4j system properly
It seems the app is not configured properly on my machine whereas all I did is run the .exec file from the download.

**Comment by jyoshimi** [comment]
Ok got it. I will try to get to this this weekend off to work now.

**Comment by jyoshimi** [comment]
Re: log4j it's a known issue, and just a warning.  Nothing really to worry about.

**Comment by SBL2021** [comment]
Ok then the issue is the exception in thread "AWT-EventQueue-0"

**Comment by SBL2021** [comment]
Ok thx Jeff I will await for you to get back to me on this. Greatly appreciate your help.

**Comment by jyoshimi** [comment]
Yes it is that exception. The issue is with xstream, the library we use to save to and open from xml. If you looked at the "caused by" line that is where the "real" issue begin.

I was able to recreate the bug on my machine, but I am running a newer java.  I have not yet downloaded java 8.  For now, however, this might work. We have not released it yet, but are working on Simbrain 3.05, which addresses this and also packages Simbrain in a more familiar app interface.  Try that and let me know if it works.

https://simbrain.net/Downloads/downloads_temp.html

**Comment by SBL2021** [comment]
Ok let me try it & get back to you.

**Comment by SBL2021** [comment]
It worked. 
The only issue is that the install exec put it in the Temp file of the App data. Also can I use v4 of is it still unstable to use?

**Comment by SBL2021** [comment]
Different problem now
when I open a zip file from open network it does not open. Here is the log
[Fatal Error] :1:1: Content is not allowed in prolog.
Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at org.simbrain.workspace.WorkspaceSerializer.open(WorkspaceSerializer.java:360)
        at org.simbrain.workspace.actions.OpenNetworkAction.actionPerformed(OpenNetworkAction.java:67)
        at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
        at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
        at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
        at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
        at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
        at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
        at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
        at java.awt.Component.processMouseEvent(Component.java:6539)
        at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
        at java.awt.Component.processEvent(Component.java:6304)
        at java.awt.Container.processEvent(Container.java:2239)
        at java.awt.Component.dispatchEventImpl(Component.java:4889)
        at java.awt.Container.dispatchEventImpl(Container.java:2297)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
        at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
        at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
        at java.awt.Container.dispatchEventImpl(Container.java:2283)
        at java.awt.Window.dispatchEventImpl(Window.java:2746)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
        at java.awt.EventQueue$4.run(EventQueue.java:733)
        at java.awt.EventQueue$4.run(EventQueue.java:731)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.simbrain.workspace.WorkspaceSerializer.open(WorkspaceSerializer.java:352)
        ... 39 more
Caused by: com.thoughtworks.xstream.io.StreamException:
        at com.thoughtworks.xstream.io.xml.DomDriver.createReader(DomDriver.java:115)
        at com.thoughtworks.xstream.io.xml.DomDriver.createReader(DomDriver.java:84)
        at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1346)
        at org.simbrain.network.NetworkComponent.open(NetworkComponent.java:431)
        ... 44 more
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.
        at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:257)
        at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)
        at com.thoughtworks.xstream.io.xml.DomDriver.createReader(DomDriver.java:108)
        ... 47 more

**Comment by jyoshimi** [comment]
v4 should be fine  as long as you are using jdk 8. Then again you ran in to problems in that situation, but I have yet to replicate them.

Oh yeah the default storage location is app data, so you will have to manually navigate to where you want to save it if not there.

**Comment by SBL2021** [comment]
ok I will move it manually Thx

**Comment by jyoshimi** [comment]
To open saved workspace zip files use the open button at the top level.  "Open workspace".

**Comment by SBL2021** [comment]
that is what I have used

**Comment by SBL2021** [comment]
I am still getting errors
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
        at java.io.ByteArrayInputStream.<init>(ByteArrayInputStream.java:106)
        at org.simbrain.workspace.WorkspaceSerializer.deserialize(WorkspaceSerializer.java:203)
        at org.simbrain.workspace.WorkspaceSerializer.deserialize(WorkspaceSerializer.java:159)
        at org.simbrain.workspace.Workspace.openWorkspace(Workspace.java:656)
        at org.simbrain.workspace.gui.SimbrainDesktop.openWorkspace(SimbrainDesktop.java:1061)
        at org.simbrain.workspace.actions.OpenWorkspaceAction.actionPerformed(OpenWorkspaceAction.java:56)
        at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
        at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
        at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
        at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
        at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
        at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
        at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
        at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
        at java.awt.Component.processMouseEvent(Component.java:6539)
        at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
        at java.awt.Component.processEvent(Component.java:6304)
        at java.awt.Container.processEvent(Container.java:2239)
        at java.awt.Component.dispatchEventImpl(Component.java:4889)
        at java.awt.Container.dispatchEventImpl(Container.java:2297)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
        at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
        at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
        at java.awt.Container.dispatchEventImpl(Container.java:2283)
        at java.awt.Window.dispatchEventImpl(Window.java:2746)
        at java.awt.Component.dispatchEvent(Component.java:4711)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
        at java.awt.EventQueue$4.run(EventQueue.java:733)
        at java.awt.EventQueue$4.run(EventQueue.java:731)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by jyoshimi** [comment]
Hmm, you make a workspace in 3.05, save it, and re-open, and get that? I'll have to look into that later and try to replicate.

**Comment by SBL2021** [comment]
Ok strange when I opened IAC_full.zip it opened. In fact any of the .zip workspaces which come with the app open no problem. However when I try the .zip files i did create yesterday it gives me the error above. So I will try and save the workspaces with 3.04 and see if it works

**Comment by jyoshimi** [comment]
Ohhh, there might be incompatibilities between 3.04 and 3.05. Not sure, but maybe. Yes I suggest trying to make and save a workspace in 3.05

**Comment by SBL2021** [comment]
Hummm it did not work. what I do is create a zip file from the saved .xml file created from simbrain 05 (the one you sent me which is 3.04 not 3.05

**Comment by SBL2021** [comment]
ooops the one you sent me is 3.04 I do not have 3.05 so it is the only one am using

**Comment by jyoshimi** [comment]
The one from here

https://simbrain.net/Downloads/downloads_temp.html

should be 3.05. If it does not say 3.05 then that's a mistake.

**Comment by SBL2021** [comment]
It seems so. Let me re-install and rerun & get back to you

**Comment by SBL2021** [comment]
just double checked the link states 3.05 but when I open the app the header states 3.04

**Comment by SBL2021** [comment]
Also where does the install put the rest of the files other than the exec?

**Comment by jyoshimi** [comment]
Thanks good to know. I think they are all in some hidden directory, but I'll have to check and get back. We are planning to work on this next week so I hope you can wait a bit on that.

**Comment by SBL2021** [comment]
Yes it can wait greatly appreciate your help :-)

**Comment by jyoshimi** [comment]
I think a few things are happening here. First, we checked and for us  the windows Simbrain from that link does show 3.05 in its program header.  Also, it's tricky to command line output from it. So perhaps what is happening is that when you run from the command line you are just re-running your 3.04?  Second, when you opened your file from 3.04 was it the and and or xml that you zipped? You actually have to save _as_ a workspace file for the zip file to open properly.  If that does not resolve, I might have a minute to try to debug if you're open to a zoom call or something.

**Comment by SBL2021** [comment]
Ok Thx Jeff. Let me try again & get back to you.

**Comment by SBL2021** [comment]
I just reinstalled the exec of 3.05. It seems to work properly. I will let you know if I find any other road bloc. Thx again for your help!

**Comment by jyoshimi** [comment]
Great, closing the ticket for now!

**Comment by jyoshimi** [comment]
Hey there, there were bugs on the windows version.  We just uploaded a fixed version. Hopefully things start going more smoothly

**Comment by SBL2021** [comment]
Thanks Jeff! I will download the window fixed version 5

**Comment by SBL2021** [comment]
Hi Jeff  & al I am trying the backprop model on my data but I get systematically the same error of 87.45 even if I randomize several times my data. Any suggestions would be helpful. Thanks
BTW I used another tool with the same data and I was able to have an error of 0.1 to 0.3

**Comment by jyoshimi** [comment]
So you can regularly save and reopen files etc. right? That much is working?  For backprop, yeah, it's hard.  We just have vanilla backprop, and I suspect there are some imperfections in it (as your addendum just now shows).  You can try to lower the error rate, increase momentum, etc.  you can also try re-setting the weights not with the button but with w > r (select weights and randomize in the gui).  Feel free to open a ticket on this, saying what the other tool is and how to replicate the issue with a simple dataset. That will at least make the issue public.  But it might be quite a while before we get to it... In Simbrain 4 we are using a better library for backprop, that is based on tensorfllow.  So that should resolve it. But it would be nice if simple backprop in Simbrain 3 worked better.

**Comment by SBL2021** [comment]
Yes thx Jeff :-) I am still learning to use the tool but at least I could run the examples (not all of them) and I am trying to cearte my modeling. I will open a ticket once I capture all the details.
